### ISSUE TITLE
Bad type conversion if movie is unwatched and hasattr called

### ISSUE THREAD

**Issue by tdorsey** [pkkid/python-plexapi#191]
Refs #190 
```python
  File "/home/tommy/plexsync/base.py", line 16, in dump
    if hasattr(obj, attr):
  File "/usr/lib/python3.6/site-packages/plexapi/base.py", line 274, in __getattribute__
    value = utils.getattributeOrNone(PlexPartialObject, self, attr)
  File "/usr/lib/python3.6/site-packages/plexapi/utils.py", line 81, in getattributeOrNone
    return super(obj, self).__getattribute__(attr)
  File "/usr/lib/python3.6/site-packages/plexapi/video.py", line 49, in isWatched
    return bool(self.viewCount > 0)
*TypeError: '>' not supported between instances of 'NoneType' and 'int'*
```

On an unwatched movie, viewCount is None and the conversion fails.

**Comment by Hellowlol** [comment]
It should be None unless the object isnt loaded.

See https://github.com/pkkid/python-plexapi/blob/master/plexapi/video.py#L42

**Comment by pkkid** [comment]
You you break the code snippet to it's simplest form and post it here so we can understand what you were doing?

If you're still looking at converting these to json, you may want to take a look and understand [plex-listattrs.py](https://github.com/pkkid/python-plexapi/blob/master/tools/plex-listattrs.py). I essentially did a recursive walk of the objects in there. It gets tricky when trying to deal with partial objects returned in search listing and such.

**Comment by tdorsey** [comment]
Yeah, I’ll get a sample together. Essentially I’ve overridden the default json encoder for anything inheriting from PlexObject and am naively json dumping its __dict__, but that’s causing issues with recursing into things like the session state of a PlexServer object.

**Comment by Hellowlol** [comment]
I took a look at your project and i see some things that we can do to speed up your querys.
ATM you do
````
def getMedia(server, section):
    results = server.library.section(section).search()
    api_objects = []
    if section == "Movies":
        for r in results:
            a = APIObject(r)
            api_objects.append(a)
        return set(api_objects)
    elif section == "TV Shows":
        # Show objects don't have a GUID, so grab the first episode and read it from there
        # This is really inefficient, since to query a single episode, the python-plexapi creates a
        # query for all episodes, and then filters down to one
        for r in results:
            episode = next(iter(r.episodes() or []), None)
            a = APIObject(episode)
            api_objects.append(a)
        return set(api_objects)
````

Thats really slow.

````
def getMedia(section='TV Shows'):
    for item in plex.library.section(section).all():
        # guid does not exist in the xml response to it will reload once for each show.
        print(item.guid)

````

**Comment by tdorsey** [comment]
Appreciate it. I assumed that could have been written to execute faster, it was just on the back burner. However, even with your patch applied, both movies and shows are roughly the same execution speed. A low hundreds movie library takes about 40s, another with 500+ hits the 2m gateway timeout. I had assumed my use case would have been improved by being able to cache PlexObjects as JSON, but that appears to not be the case, or the problem (the raw xml is retrieved by GET in sub 2s). 

Closing this, but thanks for the help!

**Comment by Hellowlol** [comment]
Post your code in a gist or tell me how I can test it and try to help you out. That takes way to long time.

**Comment by pkkid** [comment]
Fix TypeError when viewCount is not set #192.

**Comment by tdorsey** [comment]
@Hellowlol https://github.com/tdorsey/plexsync/tree/develop code is here. There's a Dockerfile that should stand up a flask server on localhost, but you can see the speed issues when running cli.py. Config file defaults to ~/.config/plexsync/config.ini
